commands/proposal_submission.go

```diff
@@ -1092,6 +1092,44 @@ func checkNewRiskParameters(config *protoTypes.NewMarketConfiguration) Errors {
 	return errs
 }
 
+func checkSLAParams(config *protoTypes.LiquiditySLAParameters, parent string) Errors {
+	errs := NewErrors()
+	if config == nil {
+		return errs.FinalAddForProperty(fmt.Sprintf("%s.sla_params", parent), ErrIsRequired)
+	}
+
+	lppr, err := num.DecimalFromString(config.PriceRange)
+	if err != nil {
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrIsNotValidNumber)
+	} else if lppr.IsNegative() || lppr.IsZero() {
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrMustBePositive)
+	} else if lppr.GreaterThan(num.DecimalFromInt64(100)) {
+		errs.AddForProperty(fmt.Sprintf("%s.price_range", parent), ErrMustBeAtMost100)
+	}
+
+	committmentMinTimeFraction, err := num.DecimalFromString(config.CommittmentMinTimeFraction)
+	if err != nil {
+		errs.AddForProperty(fmt.Sprintf("%s.committmentMinTimeFraction", parent), ErrIsNotValidNumber)
+	} else if committmentMinTimeFraction.IsNegative() || committmentMinTimeFraction.GreaterThan(num.DecimalOne()) {
+		errs.AddForProperty(fmt.Sprintf("%s.committmentMinTimeFraction", parent), ErrMustBeWithinRange01)
+	}
+
+	if config.ProvidersFeeCalculationTimeStep == 0 {
+		errs.AddForProperty(fmt.Sprintf("%s.providers.fee.calculationTimeStep", parent), ErrMustBePositive)
+	}
+
+	slaCompetitionFactor, err := num.DecimalFromString(config.SlaCompetitionFactor)
+	if err != nil {
+		errs.AddForProperty(fmt.Sprintf("%s.slaCompetitionFactor", parent), ErrIsNotValidNumber)
+	} else if slaCompetitionFactor.IsNegative() || slaCompetitionFactor.GreaterThan(num.DecimalOne()) {
+		errs.AddForProperty(fmt.Sprintf("%s.slaCompetitionFactor", parent), ErrMustBeWithinRange01)
+	}
+
+	// TODO are there constraints for performanceHysteresisEpochs?
```

Comment:
To be don ein this PR or later? Does htat needs a spec updte?

Reply:
as far as I can see there are no constraints for the market params in the spec other than the ones enforced here, so will require a spec update and possibly adding this later.